proposals/0010-vk-buffer-ref.md

```diff
@@ -120,8 +122,12 @@ Note the operations that are not allowed:
 *   The comparison operators == and != are not supported for buffer pointers.
 
 Most of these restrictions are there for safety. They minimize the possibility
```

Comment:
The section above is not true anymore no?
Since we can now build a `BufferPointer` from a uint64_t, and case it back to uint64_t, all the operations above (arithmetic, comparison, etc) are "allowed".

Reply:
It's more I don't see why arithmetic/comparison is not allowed now, since we allow a much less safe option which is to freely convert any uint64_t to a pointer and back.

Reply:
This forces the developer to be explicit that they are using integers. The idea it to make the dangerous behaviour hard, ugly, and annoying. It also matches what is done in SPIR-V, where the pointer must be cast to an integer before these operations are done.

Reply:
> The section above is not true anymore no? Since we can now build a `BufferPointer` from a uint64_t, and case it back to uint64_t, all the operations above (arithmetic, comparison, etc) are "allowed".

They are "supported" indirectly. You are still not allowed to use "==" with two buffer pointers, so this statement is still true.

Reply:
Right, thanks for the precision. This makes sense.